lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -147,19 +143,53 @@ void _styleAutofillElements(
   elementStyle.setProperty('caret-color', 'transparent');
 }
 
+void _ensureEditingElementInView(DomElement element, int viewId) {
+  final bool isAlreadyAppended = element.isConnected ?? false;
+  if (!isAlreadyAppended) {
+    // If the element is not already appended to a view, we don't need to move
+    // it anywhere.
+    return;
+  }
+
+  final FlutterViewManager viewManager = EnginePlatformDispatcher.instance.viewManager;
+  final EngineFlutterView? currentView = viewManager.findViewForElement(element);
+  if (currentView == null) {
+    // For some reason, the input element was in the DOM, but it wasn't part of
+    // any Flutter view. Should we throw?
+    return;
+  }
+
+  if (currentView.viewId != viewId) {
+    _insertEditingElementInView(element, viewId);
+  }
+}
+
+void _insertEditingElementInView(DomElement element, int viewId) {
```

Comment:
Also, would it make sense to pass the computed `view.dom.textEditingHost` as the second parameter to these methods, rather than the viewId, so they don't depend on the `viewManager`?

Reply:
Doing that would only move the `viewManager` references to other places in the code (`DefaultTextEditingStrategy` and `EngineAutofillForm`).  So I'm not sure that there's any benefit to doing it as you suggested.

I liked the idea of keeping `DefaultTextEditingStrategy` et al operating on `viewId` alone and not deal with the views themselves.